app/src/main/java/com/example/hris/repository/HrisRepository.kt

```diff
@@ -37,34 +37,36 @@ class HrisRepository @Inject constructor(
         emailAddress: String,
         mobileNumber: String,
         landline: String?
-    ): String {
-        val call = HrisApi.retrofitService.updateProfile(
-            userData.value!!.userID,
-            firstName,
-            middleName,
-            lastName,
-            emailAddress,
-            mobileNumber,
-            landline
-        )
+    ): ResponseModel {
+        return withContext(Dispatchers.IO) {
+            val call = HrisApi.retrofitService.updateProfile(
+                userData.value!!.userID,
+                firstName,
+                middleName,
+                lastName,
+                emailAddress,
+                mobileNumber,
+                landline
+            )
 
-        if (call.isSuccess) {
-            hrisDao.deleteProfile()
-            hrisDao.insertProfile(
-                User(
-                    userData.value!!.userID,
-                    userData.value!!.idNumber,
-                    firstName,
-                    middleName,
-                    lastName,
-                    emailAddress,
-                    mobileNumber,
-                    landline
+            if (call.isSuccess) {
+                hrisDao.deleteProfile()
+                hrisDao.insertProfile(
+                    User(
+                        userData.value!!.userID,
+                        userData.value!!.idNumber,
+                        firstName,
+                        middleName,
+                        lastName,
+                        emailAddress,
+                        mobileNumber,
+                        landline
+                    )
                 )
```

Comment:
Why not just create an `@Update` method in the `HrisDao` instead of doing 2 consecutive IO calls.